src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,7 +30,14 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        // this.value = trimmedAddress;
+        
+        String[] segmentedAddress = trimmedAddress.split(",");
+        this.block = new Block(segmentedAddress[0]);
```

Comment:
Recall the best practices discussed in Tutorial 2: avoid magic numbers :-). The indices can be replaced by constants for ease of understanding / modification.